metaflow/plugins/kfp/kfp_step_function.py

```diff
@@ -102,3 +356,86 @@ def kfp_step_function(
         "StepOpRet", ["foreach_splits"] + list(preceding_component_inputs_dict.keys())
     )(*values)
     return ret
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--metaflow_run_id", type=str, required=True)
+    parser.add_argument("--metaflow_configs", type=json.loads, required=True)
+    parser.add_argument("--cd_into_metaflow_package_cmd", type=str, required=True)
+    parser.add_argument("--clean_volume_cmd", type=str, required=True)
+    parser.add_argument("--task_id", type=str, required=True)
+    parser.add_argument("--task_id_template", type=str, required=True)
+    parser.add_argument("--step_name", type=str, required=True)
+    parser.add_argument("--flow_name", type=str, required=True)
+    parser.add_argument("--namespace", action="store_true")
+    parser.add_argument("--tags", type=json.loads, required=True)
+    parser.add_argument("--need_split_index", action="store_true")
+    parser.add_argument("--environment_type", type=str, required=True)
+    parser.add_argument("--logger_type", type=str, required=True)
+    parser.add_argument("--monitor_type", type=str, required=True)
+    parser.add_argument("--user_code_retries", type=int, required=True)
+    parser.add_argument("--workflow_name", type=str, required=True)
+    parser.add_argument("--script_name", type=str, required=True)
+    parser.add_argument("--passed_in_split_indexes", type=str, required=False)
+    parser.add_argument("--preceding_component_inputs", type=json.loads, required=False)
+    parser.add_argument(
```

Comment:
`Click` is already installed with Metaflow, why not use it?

Reply:
I went with `argparse` because `KFP` uses `argparse`. I essentially followed the KFP code to design these parser arguments. I would prefer to keep as is because a lot of the parameters are nuanced (e.g. double `json.dumps`), so changing to `Click` would take time.

Reply:
Metaflow uses Click.. and it's already installed.  It's best to be consistent with the code base.